Library/Homebrew/keg_relocate.rb

```diff
@@ -177,41 +177,34 @@ def recursive_fgrep_args
 
   def egrep_args
     grep_bin = "grep"
-    grep_args = recursive_fgrep_args
-    grep_args += "Pa"
+    grep_args = "-Pla"
     [grep_bin, grep_args]
   end
   alias generic_egrep_args egrep_args
 
-  def each_unique_file(io, block)
-    hardlinks = Set.new
-
-    until io.eof?
-      file = Pathname.new(io.readline.chomp)
-      # Don't return symbolic links.
-      next if file.symlink?
+  def each_unique_file_matching(string)
+    Utils.popen_read("fgrep", recursive_fgrep_args, string, to_s) do |io|
+      hardlinks = Set.new
 
-      # To avoid returning hardlinks, only return files with unique inodes.
-      # Hardlinks will have the same inode as the file they point to.
-      block.call file if hardlinks.add? file.stat.ino
-    end
-  end
+      until io.eof?
+        file = Pathname.new(io.readline.chomp)
+        # Don't return symbolic links.
+        next if file.symlink?
 
-  def each_unique_file_matching(string, &block)
-    Utils.popen_read("fgrep", recursive_fgrep_args, string, to_s) do |io|
-      each_unique_file(io, block)
+        # To avoid returning hardlinks, only return files with unique inodes.
+        # Hardlinks will have the same inode as the file they point to.
+        yield file if hardlinks.add? file.stat.ino
+      end
     end
   end
 
-  def each_unique_binary_file(&block)
+  def binary_file?(file)
     grep_bin, grep_args = egrep_args
 
     # We need to pass NULL_BYTE_STRING, the literal string "\x00", to grep
     # rather than NULL_BYTE, a literal null byte, because grep will internally
     # convert the literal string "\x00" to a null byte.
-    Utils.popen_read(grep_bin, grep_args, NULL_BYTE_STRING, to_s) do |io|
-      each_unique_file(io, block)
-    end
+    Utils.popen_read(grep_bin, grep_args, NULL_BYTE_STRING, file).size.positive?
```

Comment:
Will this stop when it reads the first null byte?

Reply:
Yes, the "-l" argument we use on both OSes is short for "--files-with-matches", which according the manuals for GNU and BSD grep will stop scanning after the first match.  Using the long names of the arguments as you suggested above will help to clarify this.